packages/sirius-web/frontend/sirius-web-application/src/views/project-browser/create-projects-area/ProjectTemplatesModal.tsx

```diff
@@ -0,0 +1,170 @@
+/*******************************************************************************
+ * Copyright (c) 2023, 2024 Obeo.
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+
+import Dialog from '@material-ui/core/Dialog';
+import DialogContent from '@material-ui/core/DialogContent';
+import DialogTitle from '@material-ui/core/DialogTitle';
+import Typography from '@material-ui/core/Typography';
+import { makeStyles } from '@material-ui/core/styles';
+import Pagination from '@material-ui/lab/Pagination';
+import gql from 'graphql-tag';
+import { useEffect, useState } from 'react';
+import { Redirect } from 'react-router-dom';
+import { NewProjectCard, ProjectTemplateCard, UploadProjectCard } from './ProjectTemplateCard';
+import { ProjectTemplatesModalProps, ProjectTemplatesModalState } from './ProjectTemplatesModal.types';
+import { redirectUrlFromTemplate } from './redirectUrlFromTemplate';
+import { useCreateProjectFromTemplate } from './useCreateProjectFromTemplate';
+import { useProjectTemplates } from './useProjectTemplates';
+import { GQLProjectTemplate } from './useProjectTemplates.types';
+
+export const getProjectTemplatesQuery = gql`
+  query getProjectTemplates($page: Int!) {
+    viewer {
+      projectTemplates(page: $page, limit: 12) {
+        edges {
+          node {
+            id
+            label
+            imageURL
+          }
+        }
+        pageInfo {
+          count
+        }
+      }
+    }
+  }
+`;
+
+const useProjectTemplatesModalStyles = makeStyles((theme) => ({
+  content: {
+    display: 'grid',
+    gridTemplateRows: '1fr min-content',
+    gap: theme.spacing(2),
+  },
+  templateCards: {
+    display: 'grid',
+    gridTemplateColumns: '1fr 1fr 1fr',
+    gridTemplateRows: '1fr 1fr 1fr 1fr',
+  },
+  navigation: {
+    justifySelf: 'center',
+  },
+}));
+
+export const ProjectTemplatesModal = ({ onClose }: ProjectTemplatesModalProps) => {
+  const styles = useProjectTemplatesModalStyles();
+
+  const [state, setState] = useState<ProjectTemplatesModalState>({
+    page: 0,
```

Comment:
There is an issue with the paging here.
After opening the modal, you can click on the next page and it throws an error.